src/Yasumi/Provider/UnitedKingdom/NorthernIreland.php

```diff
@@ -0,0 +1,148 @@
+     * of Ireland, Northern Ireland, the Canadian province of Newfoundland and Labrador, and the British Overseas
+     * Territory of Montserrat.
+     *
+     * @link https://en.wikipedia.org/wiki/Saint_Patrick%27s_Day
+     *
+     * @throws \Yasumi\Exception\InvalidDateException
+     * @throws \InvalidArgumentException
+     * @throws \Yasumi\Exception\UnknownLocaleException
+     * @throws \Exception
+     * @throws \Exception
+     */
+    private function calculateStPatricksDay(): void
+    {
+        if ($this->year < 1971) {
+            return;
+        }
+
+        $holiday = new Holiday(
+            'stPatricksDay',
+            ['en_GB' => 'St. Patrick\'s Day'],
+            new DateTime($this->year . '-3-17', new DateTimeZone($this->timezone)),
+            $this->locale,
+            Holiday::TYPE_BANK
+        );
+
+        $this->addHoliday($holiday);
+
+        // Substitute holiday is on the next available weekday if a holiday falls on a Saturday or Sunday
+        if (\in_array((int)$holiday->format('w'), [0, 6], true)) {
+            $substituteHoliday = clone $holiday;
+            $substituteHoliday->modify('next monday');
+
+            $this->addHoliday(new Holiday(
+                'substituteHoliday:' . $substituteHoliday->shortName,
+                ['en_GB' => $substituteHoliday->getName() . ' (substitute day)'],
+                $substituteHoliday,
+                $this->locale,
+                Holiday::TYPE_BANK
+            ));
+        }
+    }
+
+    /**
+     * Battle of the Boyne.
+     *
+     * Orangemen's Day, also called The Twelfth or Glorious Twelfth) celebrates the Glorious Revolution (1688)
+     * and victory of Protestant King William of Orange over Catholic king James II at the Battle of the
+     * Boyne (1690), which began the Protestant Ascendancy in Ireland.
+     *
+     * @link https://en.wikipedia.org/wiki/The_Twelfth
+     *
+     * @throws \Yasumi\Exception\InvalidDateException
+     * @throws \InvalidArgumentException
+     * @throws \Yasumi\Exception\UnknownLocaleException
+     * @throws \Exception
+     * @throws \Exception
+     */
+    private function calculateBattleOfTheBoyne(): void
+    {
+        if ($this->year < 1926) {
+            return;
+        }
+
+        $holiday = new Holiday(
+            'battleOfTheBoyne',
+            ['en_GB' => 'Battle of the Boyne'],
+            new DateTime($this->year . '-7-12', new DateTimeZone($this->timezone)),
+            $this->locale,
+            Holiday::TYPE_BANK
+        );
+
+        $this->addHoliday($holiday);
+
+        // Substitute holiday is on the next available weekday if a holiday falls on a Saturday or Sunday
+        if (\in_array((int)$holiday->format('w'), [0, 6], true)) {
+            $substituteHoliday = clone $holiday;
+            $substituteHoliday->modify('next monday');
+
+            $this->addHoliday(new Holiday(
+                'substituteHoliday:' . $substituteHoliday->shortName,
```

Comment:
This may need to be refactored using the SubstituteHoliday class (once that PR is merged)